example/timer/odp_timer_accuracy.c

```diff
@@ -358,14 +422,41 @@ static int start_timers(test_global_t *test_global)
 		timer_param.max_tmo = test_global->opt.max_tmo_ns;
 	}
 
+	if (mode == MODE_PERIODIC) {
+		odp_timer_periodic_capability_t capa;
+
+		capa.base_freq_hz = freq;
+		capa.max_multiplier = max_multiplier;
+		capa.res_ns = res_ns;
+
+		if (odp_timer_periodic_capability(test_global->opt.clk_src, &capa) < 0) {
+			printf("Requested periodic timer capabilities are not supported.\n");
+			printf("Capabilities: min base freq %g Hz, max base freq %g Hz, "
+			       "max res %" PRIu64 " Hz\n",
+			       odp_fract_u64_to_dbl(&timer_capa.periodic.min_base_freq_hz),
+			       odp_fract_u64_to_dbl(&timer_capa.periodic.max_base_freq_hz),
+			       timer_capa.max_res.res_hz);
+			return -1;
+		}
+
+		res_ns = test_global->opt.res_ns = capa.res_ns;
+		freq = test_global->opt.freq = capa.base_freq_hz;
```

Comment:
Related to Petri's comment, if the base frequency changes here, the original requested frequency should still be used  `run_test()` to detect drift.